catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/federation/impl/SortedQueryMonitor.java

```diff
@@ -226,6 +236,8 @@ public void run() {
 
     returnResults.setHits(totalHits);
     returnResults.addResults(sortedResults(resultList, resultComparator), true);
+
+    ThreadContext.remove();
```

Comment:
should this be done in a finally block just to make sure it always gets executed?

Reply:
The block at the top of the method will always reset it so it is less of a concern (assuming it isn't null) but probably a good idea just in case.